pkg/skaffold/deploy/kubectl/matcher/kind.go

```diff
@@ -0,0 +1,64 @@
+/*
+Copyright 2019 The Skaffold Authors
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package matcher
+
+import (
+	"strings"
+
+	"github.com/sirupsen/logrus"
+)
+
+const (
+	kindKey = "kind"
+)
+
+// Kind matches the value of "Kind" key with values not listed in the forbiddenValues.
+type Kind struct {
```

Comment:
I would rename this to `FieldMatcher` and pass in the key instead of making it only for `kind`

Reply:
agreed. we can do it. The reason i didn't make it generic was
- For it be truly generic, we will have to define the list of ignored value as `interface{}` to support others.
  where as for kind, we know the value of the kind is `string`
```
type GenericIgnoreField struct {
key string
values []interface
}
```
We have an option to only support keys with values of type `string`. 
I would like to keep it simple and support only key with `string` values for now. 

.